internal/services/dispatcher/v1/server.go

```diff
@@ -168,12 +168,24 @@ func (d *DispatcherServiceImpl) ListenForDurableEvent(server contracts.V1Dispatc
 			return fmt.Errorf("could not find external id for task %d, signal key %s", e.TaskID, e.EventKey.String)
 		}
 
+		payload, err := d.repo.Payloads().Retrieve(ctx, v1.RetrievePayloadOpts{
```

Comment:
Looks like an `n+1` - ideally we could do this out of `d.repo.Tasks().ListSignalCompletedEvents` with a bulk read.

Reply:
good call - let me change this V1TaskEvent struct to be V1TaskEventWithPayload like we have for the task and refactor a bit